cranelift/codegen/src/isa/x64/lower/isle.rs

```diff
@@ -519,6 +539,42 @@ where
     fn intcc_to_cc(&mut self, intcc: &IntCC) -> CC {
         CC::from_intcc(*intcc)
     }
+
+    #[inline]
+    fn sum_fits_in_32_bits(&mut self, offset: Offset32, constant_value: Imm64) -> Option<u32> {
+        let offset: i64 = offset.into();
+        let constant_value: i64 = constant_value.into();
+        // Sum up the two operands.
+        let sum = offset.wrapping_add(constant_value);
+        // Check that the sum will fit in 32-bits (and also is not negative).
+        if sum == ((sum << 32) >> 32) {
+            Some(sum as u32)
+        } else {
+            None
+        }
+    }
+
+    #[inline]
+    fn sum_extend_fits_in_32_bits(
+        &mut self,
+        offset: Offset32,
+        extend_from_ty: Type,
+        constant_value: Imm64,
+    ) -> Option<u32> {
+        let offset: i64 = offset.into();
+        let constant_value: u64 = constant_value.bits().try_into().unwrap();
```

Comment:
`bits()` returns an `i64`, so won't the `.try_into().unwrap()` panic if given a negative value in the `Imm64`? I think since we're working with raw bits here then determining possibility of representation below, we can cast the raw bits with `as u64`.

Reply:
Ok, changed.